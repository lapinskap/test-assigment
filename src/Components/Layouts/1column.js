import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';

export default function OneColumns({
  children, sm, md, lg,
}) {
  const result = [];
  for (let i = 0; i < children.length; i += 1) {
    const el = children[i];
    result.push(
      <Row key={i}>
        <Col sm={Array.isArray(sm) ? sm[0] : sm} md={Array.isArray(md) ? md[0] : md} lg={Array.isArray(lg) ? lg[0] : lg}>
          {el}
        </Col>
      </Row>,
    );
  }

  return result;
}

OneColumns.propTypes = {
  children: PropTypes.node.isRequired,
  lg: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  md: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  sm: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
};

OneColumns.defaultProps = {
  lg: null,
  md: null,
  sm: null,
};
