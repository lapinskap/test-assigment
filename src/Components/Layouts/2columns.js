import React from 'react';
import { Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';

export default function TwoColumns({
  children, sm, md, lg,
}) {
  const result = [];
  for (let i = 0; i < children.length; i += 2) {
    const el1 = children[i];
    const el2 = children[i + 1];
    result.push(
      <Row key={i}>
        <Col sm={Array.isArray(sm) ? sm[0] : sm} md={Array.isArray(md) ? md[0] : md} lg={Array.isArray(lg) ? lg[0] : lg}>{el1}</Col>
        <Col sm={Array.isArray(sm) ? sm[1] : sm} md={Array.isArray(md) ? md[1] : md} lg={Array.isArray(lg) ? lg[1] : lg}>{el2}</Col>
      </Row>,
    );
  }

  return result;
}

TwoColumns.propTypes = {
  children: PropTypes.node.isRequired,
  lg: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  md: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  sm: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
};

TwoColumns.defaultProps = {
  lg: null,
  md: null,
  sm: null,
};
