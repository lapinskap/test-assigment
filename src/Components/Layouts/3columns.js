import React from 'react';
import { Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';

export default function ThreeColumns({
  children, sm, md, lg,
}) {
  const result = [];
  for (let i = 0; i < children.length; i += 3) {
    const el1 = children[i];
    const el2 = children[i + 1];
    const el3 = children[i + 2];
    result.push(
      <Row key={i}>
        <Col sm={Array.isArray(sm) ? sm[0] : sm} md={Array.isArray(md) ? md[0] : md} lg={Array.isArray(lg) ? lg[0] : lg}>{el1}</Col>
        <Col sm={Array.isArray(sm) ? sm[1] : sm} md={Array.isArray(md) ? md[1] : md} lg={Array.isArray(lg) ? lg[1] : lg}>{el2}</Col>
        <Col sm={Array.isArray(sm) ? sm[2] : sm} md={Array.isArray(md) ? md[2] : md} lg={Array.isArray(lg) ? lg[2] : lg}>{el3}</Col>
      </Row>,
    );
  }

  return result;
}

ThreeColumns.propTypes = {
  children: PropTypes.node.isRequired,
  lg: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  md: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
  sm: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]),
};

ThreeColumns.defaultProps = {
  lg: null,
  md: null,
  sm: null,
};
