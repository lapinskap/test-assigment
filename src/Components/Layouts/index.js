import React from 'react';
import OneColumn from './1column';
import TwoColumns from './2columns';
import ThreeColumns from './3columns';

export const LAYOUT_TWO_COLUMNS = '2columns';
export const LAYOUT_THREE_COLUMNS = '3columns';
export const LAYOUT_ONE_COLUMN = '1column';

export const getLayout = (layout, children, layoutConfig = {}, key = 0, border = false) => {
  let LayoutComponent = null;

  if (!children.length) {
    return null;
  }

  switch (layout) {
    case LAYOUT_ONE_COLUMN:
      LayoutComponent = OneColumn;
      break;
    case LAYOUT_TWO_COLUMNS:
      LayoutComponent = TwoColumns;
      break;
    case LAYOUT_THREE_COLUMNS:
      LayoutComponent = ThreeColumns;
      break;
    default:
      console.error('Unknown layout name');
      return null;
  }

  const component = (
    <LayoutComponent key={key} sm={layoutConfig.sm} md={layoutConfig.md} lg={layoutConfig.lg}>
      {children}
    </LayoutComponent>
  );

  return border ? (
    <div className="border p-2 mt-1 mb-1" key={key}>
      {component}
    </div>
  ) : component;
};
