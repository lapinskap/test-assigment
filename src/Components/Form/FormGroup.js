import PropTypes from 'prop-types';
import React from 'react';
import { CardHeader, Card, CardBody } from 'reactstrap';
// eslint-disable-next-line import/no-cycle
import FormElement from './FormElement';
import Tooltip from '../Tooltips/defaultTooltip';
import { getLayout } from '../Layouts';

const getValueFromData = (config, data) => {
  switch (config.type) {
    case 'dateRange':
    case 'numberRange':
      return { from: data[`${config.id}From`], to: data[`${config.id}To`] };
    default:
      return data[config.id];
  }
};
export const shouldDisplayElement = (depends, data, displayCondition) => {
  if (displayCondition === false) {
    return false;
  }
  if (!depends) {
    return true;
  }

  if (Array.isArray(depends)) {
    return !depends.find(({ value, field }) => data[field] !== value);
  }
  const { field, value, functionValidation } = depends;
  if (functionValidation) {
    return functionValidation(data);
  }
  return data[field] === value;
};

const formElementsMapper = (formElementConfig, key, data, validateField, invalidFields, defaultOnChange) => {
  const {
    component, onChange, depends, displayCondition, layout, layoutConfig, border, formElements: layoutFromElements, ...restConfig
  } = formElementConfig;

  if (layout) {
    return getLayout(layout, layoutFromElements.map(
      (LayoutElementConfig, elKey) => formElementsMapper(LayoutElementConfig, elKey, data, validateField, invalidFields, defaultOnChange),
    ).filter(Boolean), layoutConfig, key, border);
  }

  if (!shouldDisplayElement(depends, data, displayCondition)) {
    return null;
  }
  if (component) {
    if (restConfig.validation) {
      return React.cloneElement(component, {
        errorMessage: invalidFields[restConfig.id],
        validate: (componentData) => validateField(restConfig.id, componentData, restConfig.validation),
      });
    }
    return component;
  }
  const value = getValueFromData(formElementConfig, data);
  return (
    <FormElement
      /* eslint-disable-next-line react/no-array-index-key */
      key={key}
      value={value}
      validateField={validateField}
      errorMessage={invalidFields[restConfig.id]}
      onChange={onChange || defaultOnChange}
      /* eslint-disable-next-line react/jsx-props-no-spreading */
      {...restConfig}
    />
  );
};

export default function FormGroup({
  title,
  formElements,
  data,
  defaultOnChange,
  noCards,
  tooltip,
  id,
  groupsAsColumns,
  invalidFields,
  validateField,
}) {
  const elements = formElements.map(
    (formElementConfig, key) => formElementsMapper(formElementConfig, key, data, validateField, invalidFields, defaultOnChange),
  );
  switch (true) {
    case noCards:
      return <NoCardWrapper title={title} tooltip={tooltip} id={id}>{elements}</NoCardWrapper>;
    case groupsAsColumns:
      return <NoCardWrapper title={title} tooltip={tooltip} id={id}>{elements}</NoCardWrapper>;
    default:
      return <CardWrapper title={title} tooltip={tooltip} id={id}>{elements}</CardWrapper>;
  }
}

const CardWrapper = ({
  title, children, tooltip, id,
}) => {
  let tooltipComponent;
  if (tooltip) {
    tooltipComponent = <Tooltip id={`form_group_tooltip_${id}`} type={tooltip.type} content={tooltip.content} placement={tooltip.placement} />;
  }
  return (

    <Card className="main-card mb-3">
      {title ? (
        <CardHeader>
          {title}
          {tooltipComponent}
        </CardHeader>
      ) : null}
      <CardBody className="pt-4">
        {children}
      </CardBody>
    </Card>
  );
};

const NoCardWrapper = ({
  title, children, tooltip, id,
}) => {
  let tooltipComponent;
  if (tooltip) {
    tooltipComponent = <Tooltip id={`form_group_tooltip_${id}`} type={tooltip.type} content={tooltip.content} placement={tooltip.placement} />;
  }
  return (

    <div className="main-card mb-3">
      {title ? (
        <CardHeader>
          {title}
          {tooltipComponent}
        </CardHeader>
      ) : null}
      <div className="pt-4">
        {children}
      </div>
    </div>
  );
};

CardWrapper.propTypes = {
  children: PropTypes.node,
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  tooltip: PropTypes.shape({
    content: PropTypes.node,
    type: PropTypes.string,
    placement: PropTypes.string,
  }),
};

NoCardWrapper.propTypes = {
  children: PropTypes.node,
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  tooltip: PropTypes.shape({
    content: PropTypes.node,
    type: PropTypes.string,
    placement: PropTypes.string,
  }),
};

FormGroup.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  data: PropTypes.any,
  defaultOnChange: PropTypes.func,
  // eslint-disable-next-line react/forbid-prop-types
  formElements: PropTypes.array,
  id: PropTypes.number,
  noCards: PropTypes.bool,
  title: PropTypes.string,
  tooltip: PropTypes.shape({
    content: PropTypes.node,
    type: PropTypes.string,
  }),
  validateField: PropTypes.func,
  // eslint-disable-next-line react/forbid-prop-types
  invalidFields: PropTypes.object,
  groupsAsColumns: PropTypes.bool,
};

CardWrapper.defaultProps = {
  children: null,
  title: '',
  tooltip: null,
};

NoCardWrapper.defaultProps = {
  children: null,
  title: '',
  tooltip: null,
};

FormGroup.defaultProps = {
  data: {},
  defaultOnChange: () => {
  },
  formElements: [],
  id: 0,
  noCards: false,
  title: '',
  tooltip: null,
  validateField: null,
  invalidFields: null,
  groupsAsColumns: false,
};
