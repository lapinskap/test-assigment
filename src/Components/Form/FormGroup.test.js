import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FormGroup from './FormGroup';
import '../../tests/setupTests';
import { LAYOUT_TWO_COLUMNS } from '../Layouts';

describe('<FormGroup />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  // let wrapper;

  let title;
  let data;
  let invalidFields;
  let validateField;
  let formElements;

  beforeEach(() => {
    title = 'Test title';
    data = jest.fn();
    invalidFields = {};
    validateField = () => {
    };
    formElements = [{ push: jest.fn() }];
    // wrapper = shallow(<FormGroup
    //   title={title}
    //   formElements={formElements}
    //   data={data}
    //   invalidFields={invalidFields}
    //   validateField={validateField}
    // />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(<FormGroup
        title={title}
        formElements={formElements}
        data={data}
        invalidFields={invalidFields}
        validateField={validateField}
      />)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });

  it('renders correctly with valid form elements', () => {
    formElements = [{
      id: 'fullName',
      dataOldSk: 'fullName',
      label: 'Pełna nazwa firmy',
      type: 'text',
      validation: ['required', {
        method: 'minLength',
        args: [3],
      }],
    }, {
      id: 'shortName',
      dataOldSk: 'shortName',
      label: 'Nazwa skrócona',
      type: 'text',
      validation: ['required', {
        method: 'minLength',
        args: [3],
      }],
    }, {
      id: 'street',
      dataOldSk: 'street',
      label: 'Ulica',
      type: 'text',
      validation: [{
        method: 'minLength',
        args: [3],
      }],
    }, {
      id: 'postcode',
      dataOldSk: 'postalCode',
      label: 'Kod pocztowy',
      type: 'text',
    }, {
      id: 'city',
      dataOldSk: 'city',
      label: 'Miasto',
      type: 'text',
    }, {
      id: 'nip',
      dataOldSk: 'nip',
      label: 'NIP',
      type: 'text',
      valueFormatter: 'only_digits',
      validation: [{
        method: 'minLength',
        args: [9],
      }],
    }, {
      id: 'phone',
      dataOldSk: 'phoneNumber',
      label: 'Telefon',
      type: 'text',
      valueFormatter: 'only_digits',
      validation: [{
        method: 'minLength',
        args: [9],
      }],
    }, {
      id: 'fax',
      dataOldSk: 'faxNumber',
      label: 'Fax',
      type: 'text',
      valueFormatter: 'only_digits',
      validation: [{
        method: 'minLength',
        args: [9],
      }],
    }, {
      id: 'industry',
      dataOldSk: 'industry',
      label: 'Branża',
      type: 'select',
      options: [{
        value: 'industry1',
        label: 'Branża 1',
      }, {
        value: 'industry2',
        label: 'Branża 2',
      }, {
        value: 'industry3',
        label: 'Branża 3',
      }],
    }, {
      id: 'companyType',
      dataOldSk: 'employer',
      label: 'Rodzaj firmy',
      type: 'radio',
      options: [{
        value: 'employer1',
        label: 'Klient-pracodawca',
      }],
    }, {
      id: 'additionalInfo',
      dataOldSk: 'additionalInfo',
      label: 'Informacje dodatkowe',
      type: 'textarea',
    }, {
      id: 'companyFeatureTypes',
      dataOldSk: 'companyFeatureTypes',
      label: 'Funkcjonalności',
      type: 'checkbox',
      options: [
        {
          value: 'option1',
          label: 'Funkcjonalnośc 1',
        },
        {
          value: 'option2',
          label: 'Funkcjonalnośc 2',
        },
        {
          value: 'option3',
          label: 'Funkcjonalnośc 3',
        },
        {
          value: 'option4',
          label: 'Funkcjonalnośc 4',
        },
        {
          value: 'option5',
          label: 'Funkcjonalnośc 5',
        },
        {
          value: 'option6',
          label: 'Funkcjonalnośc 6',
        },
        {
          value: 'option7',
          label: 'Funkcjonalnośc 7',
        },
        {
          value: 'option8',
          label: 'Funkcjonalnośc 8',
        },
        {
          value: 'option9',
          label: 'Funkcjonalnośc 9',
        },
        {
          value: 'option10',
          label: 'Funkcjonalnośc 10',
        },
        {
          value: 'option11',
          label: 'Funkcjonalnośc 11',
        },
        {
          value: 'option12',
          label: 'Funkcjonalnośc 12',
        },
      ],
    }, {
      id: 'activationDelayDays',
      dataOldSk: 'postponeActivationValue',
      label: 'Wstrzymanie aktywacji przez (w dniach)',
      type: 'number',
    }, {
      layout: LAYOUT_TWO_COLUMNS,
      formElements: [
        {
          id: 'layoutInput1',
          label: 'Layout input 1',
          type: 'text',
        },
        {
          id: 'layoutInput2',
          label: 'Layout input 2',
          type: 'text',
        },
        {
          id: 'layoutInput3',
          label: 'Layout input 3',
          type: 'text',
        },
        {
          id: 'layoutInput4',
          label: 'Layout input 4',
          type: 'text',
        },
      ],
    }, {
      id: 'senderEmail',
      dataOldSk: 'emailSender',
      label: 'Nadawca e-maili',
      type: 'text',
    }];
    const tree = renderer
      .create(<FormGroup
        title={title}
        formElements={formElements}
        data={data}
        invalidFields={invalidFields}
        validateField={validateField}
      />)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });

  // describe('Content input', () => {
  //   it('Should capture content correctly onChange', () => {
  //     const content = wrapper.dive().find('input').at(1);
  //     content.instance().value = 'Testing';
  //     content.simulate('change');
  //     expect(setState).toHaveBeenCalledWith('Testing');
  //   });
  // });
});
