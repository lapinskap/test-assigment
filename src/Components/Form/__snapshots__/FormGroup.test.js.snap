// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> renders correctly with props 1`] = `
<div
  className="main-card mb-3 card"
>
  <div
    className="card-header"
  >
    Test title
  </div>
  <div
    className="pt-4 card-body"
  >
    <div
      className="form-group"
    >
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1={null}
          id={null}
          name={null}
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<FormGroup /> renders correctly with valid form elements 1`] = `
<div
  className="main-card mb-3 card"
>
  <div
    className="card-header"
  >
    Test title
  </div>
  <div
    className="pt-4 card-body"
  >
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="fullNameLabel"
        htmlFor="fullName"
      >
        Pełna nazwa firmy
         
        <span
          className="text-danger"
        >
          *
        </span>
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="fullName"
          id="fullName"
          name="fullName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="shortNameLabel"
        htmlFor="shortName"
      >
        Nazwa skrócona
         
        <span
          className="text-danger"
        >
          *
        </span>
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="shortName"
          id="shortName"
          name="shortName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="streetLabel"
        htmlFor="street"
      >
        Ulica
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="street"
          id="street"
          name="street"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="postcodeLabel"
        htmlFor="postcode"
      >
        Kod pocztowy
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="postcode"
          id="postcode"
          name="postcode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="cityLabel"
        htmlFor="city"
      >
        Miasto
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="city"
          id="city"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="nipLabel"
        htmlFor="nip"
      >
        NIP
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="nip"
          id="nip"
          name="nip"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="phoneLabel"
        htmlFor="phone"
      >
        Telefon
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="phone"
          id="phone"
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="faxLabel"
        htmlFor="fax"
      >
        Fax
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="fax"
          id="fax"
          name="fax"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="industryLabel"
        htmlFor="industry"
      >
        Branża
      </label>
      <div
        className="input-group"
      >
        <select
          className="form-control"
          data-t1="industry"
          id="industry"
          name="industry"
          onChange={[Function]}
        >
          <option
            value=""
          >
            ---Nie wybrano ---
          </option>
          <option
            value="industry1"
          >
            Branża 1
          </option>
          <option
            value="industry2"
          >
            Branża 2
          </option>
          <option
            value="industry3"
          >
            Branża 3
          </option>
        </select>
      </div>
    </div>
    <div
      className="input-group-omb form-group"
    >
      <label
        className=""
        data-t1="companyTypeLabel"
        htmlFor="companyType"
      >
        Rodzaj firmy
         
      </label>
      <div
        data-t1="companyType"
      >
        <div
          className="custom-radio custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="employer1"
            id="employer1"
            name="companyType"
            onChange={[Function]}
            type="radio"
            value="employer1"
          />
          <label
            className="custom-control-label"
            htmlFor="employer1"
          >
            Klient-pracodawca
          </label>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="additionalInfoLabel"
        htmlFor="additionalInfo"
      >
        Informacje dodatkowe
         
        
      </label>
      <div
        className="input-group"
      >
        <textarea
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="additionalInfo"
          id="additionalInfo"
          name="additionalInfo"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          value=""
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="companyFeatureTypesLabel"
        htmlFor="companyFeatureTypes"
      >
        Funkcjonalności
         
      </label>
      <div
        data-t1="companyFeatureTypes"
      >
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option1"
            name="companyFeatureTypes_option1"
            onChange={[Function]}
            type="checkbox"
            value="option1"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option1"
          >
            Funkcjonalnośc 1
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option2"
            name="companyFeatureTypes_option2"
            onChange={[Function]}
            type="checkbox"
            value="option2"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option2"
          >
            Funkcjonalnośc 2
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option3"
            name="companyFeatureTypes_option3"
            onChange={[Function]}
            type="checkbox"
            value="option3"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option3"
          >
            Funkcjonalnośc 3
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option4"
            name="companyFeatureTypes_option4"
            onChange={[Function]}
            type="checkbox"
            value="option4"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option4"
          >
            Funkcjonalnośc 4
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option5"
            name="companyFeatureTypes_option5"
            onChange={[Function]}
            type="checkbox"
            value="option5"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option5"
          >
            Funkcjonalnośc 5
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option6"
            name="companyFeatureTypes_option6"
            onChange={[Function]}
            type="checkbox"
            value="option6"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option6"
          >
            Funkcjonalnośc 6
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option7"
            name="companyFeatureTypes_option7"
            onChange={[Function]}
            type="checkbox"
            value="option7"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option7"
          >
            Funkcjonalnośc 7
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option8"
            name="companyFeatureTypes_option8"
            onChange={[Function]}
            type="checkbox"
            value="option8"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option8"
          >
            Funkcjonalnośc 8
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option9"
            name="companyFeatureTypes_option9"
            onChange={[Function]}
            type="checkbox"
            value="option9"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option9"
          >
            Funkcjonalnośc 9
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option10"
            name="companyFeatureTypes_option10"
            onChange={[Function]}
            type="checkbox"
            value="option10"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option10"
          >
            Funkcjonalnośc 10
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option11"
            name="companyFeatureTypes_option11"
            onChange={[Function]}
            type="checkbox"
            value="option11"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option11"
          >
            Funkcjonalnośc 11
          </label>
        </div>
        <div
          className="custom-checkbox custom-control"
          hidden={false}
        >
          <input
            aria-invalid={false}
            checked={false}
            className="custom-control-input"
            data-t1="companyFeatureTypes"
            disabled={false}
            id="companyFeatureTypes_option12"
            name="companyFeatureTypes_option12"
            onChange={[Function]}
            type="checkbox"
            value="option12"
          />
          <label
            className="custom-control-label"
            htmlFor="companyFeatureTypes_option12"
          >
            Funkcjonalnośc 12
          </label>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="activationDelayDaysLabel"
        htmlFor="activationDelayDays"
      >
        Wstrzymanie aktywacji przez (w dniach)
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="activationDelayDays"
          id="activationDelayDays"
          name="activationDelayDays"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            data-t1="layoutInput1Label"
            htmlFor="layoutInput1"
          >
            Layout input 1
             
            
          </label>
          <div
            className="input-group"
          >
            <input
              aria-invalid={false}
              autoComplete="chrome-off"
              className="form-control"
              data-t1="layoutInput1"
              id="layoutInput1"
              name="layoutInput1"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder={null}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            data-t1="layoutInput2Label"
            htmlFor="layoutInput2"
          >
            Layout input 2
             
            
          </label>
          <div
            className="input-group"
          >
            <input
              aria-invalid={false}
              autoComplete="chrome-off"
              className="form-control"
              data-t1="layoutInput2"
              id="layoutInput2"
              name="layoutInput2"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder={null}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            data-t1="layoutInput3Label"
            htmlFor="layoutInput3"
          >
            Layout input 3
             
            
          </label>
          <div
            className="input-group"
          >
            <input
              aria-invalid={false}
              autoComplete="chrome-off"
              className="form-control"
              data-t1="layoutInput3"
              id="layoutInput3"
              name="layoutInput3"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder={null}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            data-t1="layoutInput4Label"
            htmlFor="layoutInput4"
          >
            Layout input 4
             
            
          </label>
          <div
            className="input-group"
          >
            <input
              aria-invalid={false}
              autoComplete="chrome-off"
              className="form-control"
              data-t1="layoutInput4"
              id="layoutInput4"
              name="layoutInput4"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder={null}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        data-t1="senderEmailLabel"
        htmlFor="senderEmail"
      >
        Nadawca e-maili
         
        
      </label>
      <div
        className="input-group"
      >
        <input
          aria-invalid={false}
          autoComplete="chrome-off"
          className="form-control"
          data-t1="senderEmail"
          id="senderEmail"
          name="senderEmail"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={null}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
